aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/UpdateHandler.java

```diff
@@ -64,6 +64,14 @@
             updateTags(proxy, model, previousTags, tags, logger);
         }
 
+        if(!tagsChanged && !kmsKeyChanged && !retentionChanged){
+            try{
+                proxy.injectCredentialsAndInvokeV2(Translator.translateToListTagsLogGroupRequest(model.getLogGroupName()), ClientBuilder.getClient()::listTagsLogGroup);
```

Comment:
How would we reach this code? It seems like tags, kms key and retention are the only mutable properties, so if the update handler were invoked, one of these properties would have been mutated. I think it would make more sense just to catch this exception in the associated api calls for these properties.

Reply:
I don't think we can actually reach this code in real life, I've added new contract test inputs that ensure the update handler will receive requests that actually modify the resource which should prevent it from needing these additional lines